SmartCar/src/Basic_Obstacle_Avoidance.ino

```diff
@@ -0,0 +1,24 @@
+#include <Smartcar.h>
+
+BrushedMotor leftMotor(smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+const int TRIGGER_PIN=5;
+const int ECHO_PIN=18;
+const unsigned int MAX_DISTANCE = 100;
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+void setup(){
+  Serial.begin(9600);
+  }
+
+void loop() {
+  while(front.getDistance()==0 || front.getDistance() > 15){
```

Comment:
This conducts **two** measurements that may be different. If for example the first measurement is 100 and the next one is 0 (which may very well happen if there aren't any obstacles nearby) then your car will stop.
Additionally a 0 may be returned if there's something too close.
Finally, why have two loops inside each other? loop() already... Loops!

In other words there are 3 issues at line 19.:)